async-nats/src/lib.rs

```diff
@@ -11,41 +11,50 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-//! A Rust async bleeding edge client for the NATS.io ecosystem.
+//! A Rust asynchronous "bleeding edge" client for the NATS.io ecosystem.
 //!
-//! `git clone https://github.com/nats-io/nats.rs`
+//! To access the repository, you can clone it by running:
 //!
-//! NATS.io is a simple, secure and high performance open source messaging
-//! system for cloud native applications, `IoT` messaging, and microservices
+//! ```bash
+//! git clone https://github.com/nats-io/nats.rs
+//! ````
+//! NATS.io is a simple, secure, and high-performance open-source messaging
+//! system designed for cloud-native applications, IoT messaging, and microservices
 //! architectures.
 //!
-//! For sync API refer <https://crates.io/crates/nats>
+//! For the synchronous API, you can refer to:
+//! <https://crates.io/crates/nats>
 //!
-//! For more information see [https://nats.io/].
-//!
-//! [https://nats.io/]: https://nats.io/
+//! For more information on NATS.io visit: <https://nats.io>
 //!
 //! ## Examples
 //!
-//! Below you can find some basic examples how to use this library.
+//! Below, you can find some basic examples on how to use this library.
 //!
-//! For details, refer docs for specific method/struct.
+//! For more details, please refer to the docs for the specific methods and structures.
 //!
 //! ### Complete example
 //!
+//! Connect to the NATS server, publish messages and subscribe to receive messages.
+//!
 //! ```
 //! use bytes::Bytes;
 //! use futures::StreamExt;
 //!
 //! #[tokio::main]
 //! async fn main() -> Result<(), async_nats::Error> {
+//!     // Connect to the NATS server
 //!     let client = async_nats::connect("demo.nats.io").await?;
+//!
+//!     // Subscribe to the "messages" subject and limit to 10 messages
 //!     let mut subscriber = client.subscribe("messages".into()).await?.take(10);
```

Comment:
Remove the `take` - it's more confusing than helping.

Reply:
Agreed! Also, done!